### PR TITLE
Create meteo-qt.profile

### PR DESCRIPTION
**Pull request by jose1711**

### CODE REVIEW

File: etc/meteo-qt.profile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Firejail profile for meteo-qt
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Add a description:
`# Description: System tray application for weather status information`

File: etc/meteo-qt.profile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# This file is overwritten after every install/update
+# Persistent local customizations
+include meteo-qt.local
+# Persistent global definitions
+include globals.local
+
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: For consistency with existing profiles, add a comment:
`# Allow python (blacklisted by disable-interpreters.inc)`

File: etc/meteo-qt.profile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+noblacklist ${PATH}/python3*
+noblacklist /usr/lib/python3*
+
+include disable-common.inc
+include disable-devel.inc
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Add newly introduced disable-exec.inc
`include disable-exec.inc`

File: etc/meteo-qt.profile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Move above # Allow python ...

File: etc/meteo-qt.profile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+noblacklist ${HOME}/.config/meteo-qt
+blacklist ${HOME}/.config/*
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Remove 'blacklist ${HOME}/.config/*'

File: etc/meteo-qt.profile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Make this a whitelist profile by adding
```
mkdir ${HOME}/.config/meteo-qt
whitelist ${HOME}/.config/meteo-qt
include whitelist-common.inc
include whitelist-var-common.inc
```

File: etc/meteo-qt.profile [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+nogroups
+nonewprivs
+noroot
+nosound
+notv
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: No 'tracelog'?

File: etc/meteo-qt.profile [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+disable-mnt
+private-cache
+private-dev
+private-tmp
+
+memory-deny-write-execute
```
Comment (glitsj16) [inline etc/meteo-qt.profile]: Remove both noexec lines, these are now handled in disable-exec.inc

**Comment by jose1711** [comment]
thank you for your suggestions. hopefully i addresses them all. i also added a fix for .desktop file not being created/removed from autostart

File: etc/meteo-qt.profile [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+# Description: System tray application for weather status information
+# This file is overwritten after every install/update
+# Persistent local customizations
+include meteo-qt.local
+# Persistent global definitions
+include globals.local
+
+# Allow python (blacklisted by disable-interpreters.inc)
+noblacklist ${PATH}/python3*
+noblacklist /usr/lib/python3*
```
Comment (rusty-snake) [inline etc/meteo-qt.profile]: `noblacklist /usr/local/lib/python3*`

File: etc/meteo-qt.profile [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+seccomp
+shell none
+
+disable-mnt
+private-cache
+private-dev
+private-tmp
+
+memory-deny-write-execute
+tracelog
```
Comment (rusty-snake) [inline etc/meteo-qt.profile]: Move under shell none

File: etc/meteo-qt.profile [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+noroot
+nosound
+notv
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
+
+disable-mnt
```
Comment (rusty-snake) [inline etc/meteo-qt.profile]: What about `private-bin meteo-qt,python3 (and more if needed)`?

File: etc/meteo-qt.profile [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+mkdir ${HOME}/.config/meteo-qt
+whitelist ${HOME}/.config/meteo-qt
+whitelist ${HOME}/.config/autostart
+include whitelist-common.inc
+include whitelist-var-common.inc
+
+caps.drop all
+netfilter
+nodbus
```
Comment (rusty-snake) [inline etc/meteo-qt.profile]: Doesn't break tray-icon?

Comment (jose1711) [reply etc/meteo-qt.profile]: tray icon is working fine (for me at least)

File: etc/meteo-qt.profile (hunk 11 above)
Comment (glitsj16) [reply etc/meteo-qt.profile]: Works on Arch with `private-bin meteo-qt,python*`.

File: etc/meteo-qt.profile (hunk 12 above)
Comment (glitsj16) [reply etc/meteo-qt.profile]: Confirming tray icon works fine with `nodbus`.

**Comment by glitsj16** [comment]
@jose1711 Thanks for making the requested changes. Almost ready to merge. Things left to do:
- add `noblacklist ${HOME}/.config/meteo-qt` after line 7 (needed to counteract blacklisting in disable-programs.inc)
- remove lines 14, 15 and 27 (these are never needed)
- move line 45 (memory-deny-write-execute) to the very bottom
- move line 52 under disable-mnt and use this format: private-bin meteo-qt,python*

Here's an [example](https://gist.github.com/glitsj16/db3360b8a7b1feaf01d698c3c76e90c3) of what I mean.

**Comment by jose1711** [comment]
>     * remove lines 14, 15 and 27 (these are never needed)

removing line 27 results in an error when toggling autostart option:

```
  File "/usr/lib/python3.7/site-packages/meteo_qt/settings.py", line 434, in autostart_apply
    with open(total_path, 'w') as out_file:
FileNotFoundError: [Errno 2] No such file or directory: '/home/jose/.config/autostart/meteo-qt.desktop'
```

**Comment by glitsj16** [comment]
> removing line 27 results in an error when toggling autostart option:

My bad, I see now what meteo-qt is doing. Leave in line 27 and we're set to merge.

**Comment by jose1711** [comment]
> > removing line 27 results in an error when toggling autostart option:
> 
> My bad, I see now what meteo-qt is doing. Leave in line 27 and we're set to merge.

actually, those other `noblacklist` lines are required too. before i do another push - the ordering, does it have a special meaning or it's only cosmetics?

**Comment by glitsj16** [comment]
@jose1711 Their might be some confusion here due to the many comments/requested changes. I've adapted the [example](https://gist.github.com/glitsj16/db3360b8a7b1feaf01d698c3c76e90c3) profile I mentioned above to incorporate your latest changes. Besides the python noblacklists meteo-qt only needs 2 additional noblacklist lines:
- noblacklist ${HOME}/.config/autostart (needed for the autostart functionality in the settings)
- noblacklist ${HOME}/.config/meteo-qt (main config dir)

There should be no need for `noblacklist /usr/share/applications/meteo-qt.desktop` (and additional whitelisting to make this a tighter whitelist profile) as that path is not blacklisted in any of the disable-*.inc files included in the profile. Contrary to the desktop item in ~/.config/autostart meteo-qt does not touch its desktop file under /usr/share/applications once it's started.

Regarding the ordering (both amongst options and inside options that contain multiple items like private-bin) that's common practice yes.

Hope this helps to clear up any questions you might have.

**Comment by jose1711** [comment]
thank you, i pushed your latest version. though not sure if i am even entitled to raise this PR - you did most of the work :-) i also have a lot more questions - is there an IRC, discord or similar channel where ad hoc things may be discussed?

**Comment by glitsj16** [comment]
@jose1711 Thanks for all your edits. Don't worry about entitlements or the like, this project is very much a real community effort. If not all, I certainly have been in your position too and could count on the assistance of more experienced collaborators. I don't know of any dedicated IRC channel or the likes alas. Until such avenues exist I guess you're very much bound to open issues here. Which shouldn't be any problem, they tend to receive attention quite rapidly. Looking forward to your contributions!

**Pull request merged**